pkgs/applications/misc/sioyek/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, installShellFiles
+, fetchFromGitHub
+, gumbo
+, harfbuzz
+, jbig2dec
+, mupdf
+, openjpeg
+, qt3d
+, qtbase
+, qmake
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sioyek";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "ahrm";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-G4iZi6xTJjWZN0T3lO0jPquxJ3p8Mc0ewmjJEKcGJ34=";
+  };
+
+  buildInputs = [ gumbo harfbuzz jbig2dec mupdf openjpeg qt3d qtbase ];
+
+  nativeBuildInputs = [ installShellFiles wrapQtAppsHook qmake ];
+
+  postPatch = ''
+    substituteInPlace pdf_viewer_build_config.pro \
+      --replace "-lmupdf-threads" "-lfreetype -lgumbo -ljbig2dec -lopenjp2 -ljpeg"
+    substituteInPlace pdf_viewer/main.cpp \
+      --replace "/usr/share/sioyek" "$out/share" \
+      --replace "/etc/sioyek" "$out/etc"
```

Comment:
For other reviewers: sioyek apparently also looks at /etc/xdg so this does not prevent global configuration.

Reply:
Yes, these files in /etc are default keybinds and configs, and should not be changed. 
The user configs are located at `$HOME/.local/share/sioyek/`: https://github.com/ahrm/sioyek/blob/main/pdf_viewer/main.cpp#L233
I intend to write a home manager module for this after this is merged.